lib/db.c

```diff
@@ -11162,8 +11162,10 @@ grn_ctx_at(grn_ctx *ctx, grn_id id)
                 "<%u>(<%.*s>)",
                 id,
                 name_size, name);
+            grn_db_value_unlock(ctx, id, vp);
             goto exit;
           }
+          grn_db_value_unlock(ctx, id, vp);
```

Comment:
How about the following flow to show we want to ensure unlocking?

```c
bool success = grn_db_value_wait(ctx, id, vp);
if (!success) {
  /* set error */
}
grn_db_value_unlock(ctx, id, vp); /* This sentence is always executed because there is no condition. */
if (!success) {
  goto exit;
}
```

Reply:
Or:

```c
bool success = grn_db_value_wait(ctx, id, vp);
uint32_t current_lock = vp->lock;
grn_db_value_unlock(ctx, id, vp); /* This sentence is always executed because there is no condition. */
if (!success) {
  /* set error */
  goto exit;
}
```

Reply:
Thank you for your comment.
I've fixed it at https://github.com/groonga/groonga/pull/1256/commits/4154545269ac6986972f74114f28bfc81f94554d.